### PR TITLE
ROE-1550: add form Trust Interuppt

### PR DESCRIPTION
**Pull request by dgolubevs**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-1550

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/trust.interrupt.controller.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+): TrustDetailPageProperties => {
+  const trustId = req.params[config.ROUTE_PARAM_TRUST_ID];
+
+  const appData = getApplicationData(req.session);
+
+  return {
+    backLinkUrl: `${config.TRUST_ENTRY_URL}/${trustId}${config.TRUST_INVOLVED_URL}`,
+    templateName: config.TRUST_INTERRUPT_PAGE,
+    pageParams: {
+      title: TRUST_INTERUPPT_TEXTS.title,
```
Comment (JoshFen95) [inline src/controllers/trust.interrupt.controller.ts:34]: typo referenced above

File: src/controllers/trust.interrupt.controller.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    logger.errorRequest(req, error);
+
+    return next(error);
+  }
+};
+
+export {
+  get,
+  post,
+  TRUST_INTERUPPT_TEXTS,
```
Comment (JoshFen95) [inline src/controllers/trust.interrupt.controller.ts:80]: typo referenced above

File: src/routes/index.ts [hunk 3]
```diff
@@ -36,6 +36,7 @@ import {
   trustInvolved,
   trustHistoricalbeneficialOwner,
   trustIndividualbeneficialOwner,
+  trustInterupt,
```
Comment (JoshFen95) [inline src/routes/index.ts:39]: trustInterrupt*

File: test/controllers/trust.need.submit.controller.spec.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+import { TRUST_WITH_ID } from '../__mocks__/session.mock';
+import { ANY_MESSAGE_ERROR, PAGE_TITLE_ERROR } from '../__mocks__/text.mock';
+import { TRUST_ENTRY_URL, TRUST_INTERRUPT_PAGE, TRUST_INTERRUPT_URL } from '../../src/config';
+import { get, post, TRUST_INTERUPPT_TEXTS } from '../../src/controllers/trust.interrupt.controller';
+import { authentication } from '../../src/middleware/authentication.middleware';
+import { hasTrust } from '../../src/middleware/navigation/has.trust.middleware';
+import { getApplicationData } from '../../src/utils/application.data';
+import { mapCommonTrustDataToPage } from '../../src/utils/trust/common.trust.data.mapper';
+
+describe('Trust Interuppt controller', () => {
```
Comment (JoshFen95) [inline test/controllers/trust.need.submit.controller.spec.ts:26]: Interrupt

File: src/controllers/trust.interrupt.controller.ts (hunk 1 above)
Comment (dgolubevs) [reply src/controllers/trust.interrupt.controller.ts:34]: fixed

File: src/controllers/trust.interrupt.controller.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import { NextFunction, Request, Response } from 'express';
+import * as config from '../config';
+import { logger } from '../utils/logger';
+import { getApplicationData } from '../utils/application.data';
+import { mapCommonTrustDataToPage } from '../utils/trust/common.trust.data.mapper';
+import * as PageModel from '../model/trust.page.model';
+
+const TRUST_INTERUPPT_TEXTS = {
```
[the remark this answers is not in the document]
Comment (dgolubevs) [reply src/controllers/trust.interrupt.controller.ts:8]: fixed

File: src/controllers/index.ts [hunk 6]
```diff
@@ -30,6 +30,7 @@ export * as trustDetails from "./trust.details.controller";
 export * as trustInvolved from "./trust.involved.controller";
 export * as trustHistoricalbeneficialOwner from "./trust.historical.beneficial.owner.controller";
 export * as trustIndividualbeneficialOwner from "./trust.individual.beneficial.owner.controller";
+export * as trustInterupt from "./trust.interrupt.controller";
```
[the remark this answers is not in the document]
Comment (dgolubevs) [reply src/controllers/index.ts:33]: fixed

File: src/controllers/trust.interrupt.controller.ts (hunk 2 above)
Comment (dgolubevs) [reply src/controllers/trust.interrupt.controller.ts]: fix

File: src/routes/index.ts (hunk 3 above)
Comment (dgolubevs) [reply src/routes/index.ts]: fix

File: test/controllers/trust.need.submit.controller.spec.ts (hunk 4 above)
Comment (dgolubevs) [reply test/controllers/trust.need.submit.controller.spec.ts]: fixed

**Review by JoshFen95 — APPROVED** [review]
looks good to me

**Review by okoye-companieshouse — APPROVED** [review]

**Pull request merged**